### PR TITLE
Renaming HnsNetwork based on subnet and prefix.

### PR DESCRIPTION
**Pull request by jaer-tsun**
**What this PR does / why we need it**:
Instead of basing HnsNetwork name on vlan, it's now created from subnet + prefix.
This is a temporary workaround until we can add/update HnsNetwork with multiple subnets.
e.g. azure-vlan1 -> azure-vlan1-172-28-1-0_24

### CODE REVIEW

File: cni/network/network_windows.go [hunk 1]
```diff
@@ -93,35 +93,43 @@ func setEndpointOptions(cnsNwConfig *cns.GetNetworkContainerResponse, epInfo *ne
 func addSnatInterface(nwCfg *cni.NetworkConfig, result *cniTypesCurr.Result) {
 }
 
-func updateSubnetPrefix(cnsNwConfig *cns.GetNetworkContainerResponse, subnetPrefix *net.IPNet) {
+func updateSubnetPrefix(cnsNwConfig *cns.GetNetworkContainerResponse, subnetPrefix *net.IPNet) error {
```
[the remark this answers is not in the document]
Comment (jaer-tsun) [reply cni/network/network_windows.go]: updated

File: cni/network/network_windows.go [hunk 2]
```diff
@@ -93,35 +93,43 @@ func setEndpointOptions(cnsNwConfig *cns.GetNetworkContainerResponse, epInfo *ne
...
-		subnetPrefix.IP = subnetPrefix.IP.Mask(subnetPrefix.Mask)
+		subnetPrefix.IP = ipAddr.Mask(subnetPrefix.Mask)
 		log.Printf("Updated subnetPrefix: %s", subnetPrefix.String())
 	}
+
+	return nil
 }
 
-func getNetworkName(podName, podNs, ifName string, nwCfg *cni.NetworkConfig) (string, error) {
+func getNetworkName(podName, podNs, ifName string, nwCfg *cni.NetworkConfig) (networkName string, err error) {
```
[the remark this answers is not in the document]
Comment (jaer-tsun) [reply cni/network/network_windows.go]: I changed it because it was much easier to debug with single exit at the end

File: cni/network/network_windows.go [hunk 3]
```diff
@@ -93,35 +93,43 @@ func setEndpointOptions(cnsNwConfig *cns.GetNetworkContainerResponse, epInfo *ne
...
+			*subnetPrefix = net.IPNet{Mask: net.CIDRMask(int(ipconfig.IPSubnet.PrefixLength), 32)}
+		} else if ipAddr.To16() != nil {
+			*subnetPrefix = net.IPNet{Mask: net.CIDRMask(int(ipconfig.IPSubnet.PrefixLength), 128)}
 		} else {
-			*subnetPrefix = net.IPNet{IP: ipAddr, Mask: net.CIDRMask(int(ipconfig.IPSubnet.PrefixLength), 128)}
+			return fmt.Errorf("[cni-net] Failed to get mask from CNS network configuration")
 		}
 
-		subnetPrefix.IP = subnetPrefix.IP.Mask(subnetPrefix.Mask)
+		subnetPrefix.IP = ipAddr.Mask(subnetPrefix.Mask)
```
[the remark this answers is not in the document]
Comment (jaer-tsun) [reply cni/network/network_windows.go]: assigning subnetPrefix.IP with ipAddr is unnecessary since it's going to be replaced with the masked IP

File: cni/network/network_windows.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaer-tsun) [reply cni/network/network_windows.go]: going to initialize err to nil

File: cni/network/network.go [hunk 4]
```diff
@@ -331,11 +328,11 @@ func (plugin *netPlugin) Add(args *cniSkel.CmdArgs) error {
 
 			// Derive the subnet prefix from allocated IP address.
 			subnetPrefix = result.IPs[0].Address
-
 			iface := &cniTypesCurr.Interface{Name: args.IfName}
 			result.Interfaces = append(result.Interfaces, iface)
 		}
 
+		subnetPrefix.IP = subnetPrefix.IP.Mask(subnetPrefix.Mask)
```
Comment (ashvindeodhar) [inline cni/network/network.go]: why was this moved up here?

File: cni/network/network_windows.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashvindeodhar) [reply cni/network/network_windows.go]: Let's create a func getSubnetPrefix which is called from updateSubnetPrefix. Please call getSubnetPrefix from getNetworkName as it doesn't actually updates the prefix.

Use / log the error which is returned by this function in network.go caller.
